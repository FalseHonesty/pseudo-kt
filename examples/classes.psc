class Pet
    private name

    public function new(givenName)
        name=givenName
    endfunction

    public function sayHello()
        print("Hi! My name is " + name + "!")
    endfunction

    public function getName()
        return this.name
    endfunction
endclass

class Dog inherits Pet
    private breed

    public function new(givenName, givenBreed)
        super.new(givenName)
        breed = givenBreed
    endfunction

    public function getBreed()
        return breed
    endfunction

    public function toString()
        this = null
        return "Dog(name=" + super.getName() + ", breed=" + getBreed() + ")"
    endfunction
endclass

pet = new Pet("Scruffy")
pet.sayHello()
print(pet.getName())
dog = new Dog("Fido", "Bull terrier")
dog.sayHello()
print(dog.getName())
print(dog.getBreed())
print(dog)
x = dog.getName
print(x(2))