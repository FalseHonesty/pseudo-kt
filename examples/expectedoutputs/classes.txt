Hi! My name is Scruffy!
Scruffy
Hi! My name is Fido!
Fido
Bull terrier
